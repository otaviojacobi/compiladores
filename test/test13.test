int main() {
    printf("teste\"dsada\ndsadsa"); // this is a comment and should be ignored !
    char hello = '\n';
    char list_of_chars[4] = ['a', '\\', '\n', '\''];
    int _number = /*can i do this ? "yes you should"*/ 5; 
    return 0; // My return value should be the same without comments (as long keeping same number of lines)
}