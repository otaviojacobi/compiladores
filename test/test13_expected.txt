1 TK_PR_INT [int]
1 TK_IDENTIFICADOR [main]
1 TK_ESPECIAL [(]
1 TK_ESPECIAL [)]
1 TK_ESPECIAL [{]
2 TK_IDENTIFICADOR [printf]
2 TK_ESPECIAL [(]
2 TK_LIT_STRING ["teste\"dsada\ndsadsa"]
2 TK_ESPECIAL [)]
2 TK_ESPECIAL [;]
3 TK_PR_CHAR [char]
3 TK_IDENTIFICADOR [hello]
3 TK_ESPECIAL [=]
3 TK_LIT_CHAR ['\n']
3 TK_ESPECIAL [;]
4 TK_PR_CHAR [char]
4 TK_IDENTIFICADOR [list_of_chars]
4 TK_ESPECIAL [[]
4 TK_LIT_INT [4]
4 TK_ESPECIAL []]
4 TK_ESPECIAL [=]
4 TK_ESPECIAL [[]
4 TK_LIT_CHAR ['a']
4 TK_ESPECIAL [,]
4 TK_LIT_CHAR ['\\']
4 TK_ESPECIAL [,]
4 TK_LIT_CHAR ['\n']
4 TK_ESPECIAL [,]
4 TK_LIT_CHAR ['\'']
4 TK_ESPECIAL []]
4 TK_ESPECIAL [;]
5 TK_PR_INT [int]
5 TK_IDENTIFICADOR [_number]
5 TK_ESPECIAL [=]
5 TK_LIT_INT [5]
5 TK_ESPECIAL [;]
6 TK_PR_RETURN [return]
6 TK_LIT_INT [0]
6 TK_ESPECIAL [;]
7 TK_ESPECIAL [}]